src/realm/alloc_slab.cpp

```diff
@@ -806,6 +807,20 @@ ref_type SlabAlloc::attach_file(const std::string& file_path, Config& cfg)
                 realm::util::encryption_read_barrier(map_header, 0, sizeof(Header));
             }
         }
+        if (top_ref) {
+            // Check if file size can be reduced
+            constexpr size_t file_size_ndx = 2;
+            constexpr size_t max_top_size = (file_size_ndx + 1) * 8 + sizeof(Header);
+            size_t top_page_base = top_ref & ~(page_size() - 1);
+            size_t top_offset = top_ref - top_page_base;
+            File::Map<char> map_top(m_file, top_page_base, File::access_ReadOnly, max_top_size + top_offset, 0);
+            realm::util::encryption_read_barrier(map_top, top_offset, max_top_size);
+            auto top_header = map_top.get_addr() + top_offset;
```

Comment:
It'd be much more readable to create an Array here, then use it access the element of interest.

Reply:
Cannot be done. The "reader view" is not ready yet.